test/e2e/apimachinery/flowcontrol.go

```diff
@@ -321,6 +323,7 @@ func createFlowSchema(f *framework.Framework, flowSchemaName string, matchingPre
 func makeRequest(f *framework.Framework, username string) *http.Response {
 	config := f.ClientConfig()
 	config.Impersonate.UserName = username
+	config.RateLimiter = nil
```

Comment:
hmm, since each request uses a separate config (a deep copy of the internal config), do you think we should do `f.ClientConfig()` once and pass that config around instead? alternatively, we can make ClientConfig return the right config with the rate limiter set to nil somehow

setting the rate limiter to nil on each ClientConfig, I don't think this actually removes the rate limiting

Reply:
I can't recall now, but I thought you had to set qps/burst to -1 to turn this off?

Reply:
It looks like setting this to nil will make the REST library to use the e2e framework-configured QPS limit:

https://github.com/kubernetes/kubernetes/blob/da75c266487c52425d3eb07df30229aea04bb28e/staging/src/k8s.io/client-go/rest/config.go#L333-L337

The e2e framework configured limit is 20 reqs/sec:

https://github.com/kubernetes/kubernetes/blob/540e41c825f71d5ac67bff97902c8e689784a474/test/e2e/framework/framework.go#L145

https://github.com/kubernetes/kubernetes/blob/540e41c825f71d5ac67bff97902c8e689784a474/test/e2e/framework/framework.go#L192

I couldn't find any references to the -1 thing though.

Reply:
nice catch! i crafted a new instance of rate-limiter with qps/burst set set to -1/0, the clientside rate-limiting should be bypassed now.